packages/flutter_sprites/test/image_map_test.dart

```diff
@@ -0,0 +1,72 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert' show BASE64;
+import 'dart:typed_data';
+import 'dart:ui' as ui show Image;
+
+import 'package:flutter/rendering.dart' show SynchronousFuture;
+import 'package:flutter/services.dart';
+import 'package:flutter_sprites/flutter_sprites.dart';
+import 'package:mojo/core.dart' as mojo;
+import 'package:test/test.dart';
+
+const String kTestManifest = '''
+{
+  "assets/image1.png" : [],
+  "assets/image2.png" : [],
+  "assets/image3.png" : []
+}
+''';
+
+// Base64 encoding of a 1x1 pixel png image.
+const String kTestImageBase64 =
+    'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABAQMAAAAl21bKAAAAA1BMVEUAAACnej3aAAAAAXRSTlMAQObYZgAAAApJREFUCNdjYAAAAAIAAeIhvDMAAAAASUVORK5CYII=';
+
+class TestAssetBundle extends CachingAssetBundle {
+  @override
+  Future<mojo.MojoDataPipeConsumer> load(String key) {
+    mojo.MojoDataPipe dataPipe = new mojo.MojoDataPipe();
+    final Uint8List bytes =
+        new Uint8List.fromList(BASE64.decode(kTestImageBase64));
+    dataPipe.producer.write(bytes.buffer.asByteData());
+    dataPipe.producer.handle.close();
+    return new Future<mojo.MojoDataPipeConsumer>.value(dataPipe.consumer);
+  }
+
+  @override
+  Future<String> loadString(String key, {bool cache: true}) {
+    if (key == 'AssetManifest.json')
+      return new SynchronousFuture<String>(kTestManifest);
```

Comment:
SynchronousFuture is very subtle to use properly.  It looks like the non-test loadString returns a normal Future.  We should probably do that here too.